sopel/config/types.py

```diff
@@ -696,9 +696,12 @@ def _parse(self, value, settings, section):
         if not value:
             return self.parse(value)
 
-        if value.startswith('"') or value.endswith('"'):
+        if value.startswith('"') and value.endswith('"'):
             value = value.strip('"')
 
+        if value.startswith("'") and value.endswith("'"):
+            value = value.strip("'")
```

Comment:
My first thought looking at this structure was "what if the name has multiple nested levels of quotes?"

And I thought about it a little more. This will strip `"'filename.txt'"` correctly, but `'"filename.txt"'` will only be stripped down to `"filename.txt"`.

My gut says we should either only support the first layer of quoting, or repeatedly strip both types of quote until nothing changes. I'm in favor of the first option because it's very simple to do. 😁

```suggestion
        if value.startswith('"') and value.endswith('"'):
            value = value.strip('"')
        elif value.startswith("'") and value.endswith("'"):
            value = value.strip("'")
```

Reply:
That's fair.